From 30287a29f47119b0f7280f3455abfcf95b3abf05 Mon Sep 17 00:00:00 2001
From: Jouni Malinen <quic_jouni@quicinc.com>
Date: Tue, 19 Mar 2024 16:44:00 +0200
Subject: [PATCH] WLANRTSYS-62198 Multi-AP: Use proper length for remaining
 buffer for the element

Replace the hardcoded buffer length with the actually number of
remaining bytes on the buffer. This is needed to be able to do real
buffer size validation within add_multi_ap_ie().

Furthermore, make hostapd_eid_multi_ap() static since it is not used
outside this file.

Signed-off-by: Jouni Malinen <quic_jouni@quicinc.com>
---
 src/ap/afc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/ap/afc.c b/src/ap/afc.c
index 91e684d5a..abc40a876 100644
--- a/src/ap/afc.c
+++ b/src/ap/afc.c
@@ -508,7 +508,7 @@ hostad_afc_parse_available_freq_info(struct hostapd_iface *iface,
 }
 
 
-static int hostad_afc_update_chan_info(struct afc_chan_info_elem **chan_list,
+int hostad_afc_update_chan_info(struct afc_chan_info_elem **chan_list,
 				       int *chan_list_size, u8 op_class,
 				       int center_chan, int power)
 {
-- 
2.43.0

